src/realm/realmReducer.js

```diff
@@ -303,6 +303,21 @@ export default (
           //   `op: 'update_dict'` events near the edge.)
           return state;
 
+        case EventTypes.realm_user: {
+          const { person } = event;
+
+          // TODO(flow) teach Flow that the `person.existingUser != null`s are redundant
+          if (person.is_active === false && person.existingUser != null) {
+            return { ...state, nonActiveUsers: [...state.nonActiveUsers, person.existingUser] };
+          } else if (person.is_active === true && person.existingUser != null) {
+            const userId = person.existingUser.user_id;
```

Comment:
```suggestion
            const userId = person.user_id;
```

(or further simplify)

That way we're using the `existingUser` oddity only in the cases it's needed.